evap/student/tests/test_views.py

```diff
@@ -17,64 +21,127 @@ def setUp(self):
+        cls.contributor_likert_question = mommy.make(Question, questionnaire=cls.contributor_questionnaire, type="L")
+        cls.general_text_question = mommy.make(Question, questionnaire=cls.general_questionnaire, type="T")
+        cls.general_likert_question = mommy.make(Question, questionnaire=cls.general_questionnaire, type="L")
+        cls.general_grade_question = mommy.make(Question, questionnaire=cls.general_questionnaire, type="G")
+
+        cls.contribution1 = mommy.make(Contribution, contributor=cls.contributor1, questionnaires=[cls.contributor_questionnaire],
+                                       course=cls.course)
+        cls.contribution2 = mommy.make(Contribution, contributor=cls.contributor2, questionnaires=[cls.contributor_questionnaire],
+                                       course=cls.course)
+
+        cls.course.general_contribution.questionnaires.set([cls.general_questionnaire])
+
+        cls.url = cls.url + str(cls.course.pk)
 
-    def test_complete_vote_usecase(self):
+    def fill_form(self, form, fill_complete):
+        form[question_id(self.course.general_contribution, self.general_questionnaire, self.general_text_question)] = "some text"
+        form[question_id(self.course.general_contribution, self.general_questionnaire, self.general_grade_question)] = 3
+        form[question_id(self.course.general_contribution, self.general_questionnaire, self.general_likert_question)] = 1
+
+        form[question_id(self.contribution1, self.contributor_questionnaire, self.contributor_text_question)] = "some other text"
+        form[question_id(self.contribution1, self.contributor_questionnaire, self.contributor_likert_question)] = 4
+
+        form[question_id(self.contribution2, self.contributor_questionnaire, self.contributor_text_question)] = "some more text"
+
+        if fill_complete:
+            form[question_id(self.contribution2, self.contributor_questionnaire, self.contributor_likert_question)] = 2
+
+    def test_incomplete_form(self):
         """
             Submits a student vote, verifies that an error message is
             displayed if not all rating questions have been answered and that all
-            given answers stay selected/filled and that the student cannot vote on
-            the course a second time.
+            given answers stay selected/filled.
         """
-        page = self.get_assert_200("/student/vote/5", user="lazy.student")
+        page = self.get_assert_200(self.url, user=self.voting_user1.username)
         form = page.forms["student-vote-form"]
-        form["question_17_2_3"] = "some text"
-        form["question_17_2_4"] = 1
-        form["question_17_2_5"] = 6
-        form["question_18_1_1"] = "some other text"
-        form["question_18_1_2"] = 1
-        form["question_19_1_1"] = "some more text"
-        form["question_19_1_2"] = 1
-        form["question_20_1_1"] = "and the last text"
+        self.fill_form(form, False)
```

Comment:
here and in the other calls to fill_form i would specify the parameter name `fill_complete=False`, since the code here is then easier to understand without looking at the fill_form method.